src/main/resources/jupyter/edit-mode.js

```diff
@@ -0,0 +1,337 @@
+    const syncIssueTitle = "File versions out of sync"
+    const syncIssueBody = "Your version of this file does not match the version in the workspace. What would you like to do?"
+    const syncIssueFatalBody = "This file was either deleted or never was stored with us."
+
+    //TODO URL resolution
+    const leoUrl = ''
+        // const leoUrl = 'http://localhost:8081'
+    const welderUrl = leoUrl + `/proxy/${googleProject}/${clusterName}/welder`
+    const localizeUrl = welderUrl + '/localize'
+        // const checkMetaUrl = welderUrl + '/checkMeta'
+    const checkMetaUrl = welderUrl + '/objects/metadata'
+    const lockUrl = welderUrl + '/lock'
+    const lastLockedTimer = 60000 // in ms, should be 60000 in final PR
+
+    const headers = {
+        'Content-Type': 'application/json',
+        'Access-Control-Allow-Headers': '*'
+    }
+
+    const basePayload = {
+        mode: "no-cors",
+        headers: headers
+    }
+
+    //TODO resolve these links
+    const jupyterBaseUrl = `/notebooks/${googleProject}/${clusterName}`
+    const jupyterContentsAPIUrl = jupyterBaseUrl + "/api/contents/"
+
+    const getLocalPath = () => {
+        return {
+            localObjectPath: Jupyter.notebook.notebook_path
+        }
+    }
+
+    function init() {
+        checkMeta()
+        initSyncMaintainer()
+    }
+
+    async function initSyncMaintainer() {
+        let syncMaintainer = setInterval(() => {
+            checkMeta()
+        }, lastLockedTimer)
+
+        window.onbeforeunload(function() {
+            clearInterval(syncMaintainer)
+        })
+    }
+
+    async function checkMeta() {
+        const payload = {
+            ...basePayload,
+            body: JSON.stringify(getLocalPath()),
+            method: 'POST'
+        }
+
+        return fetch(checkMetaUrl, payload)
+            .then(res => {
+                console.log('in first then of checkmeta', res)
+                if (!res.ok && res.status == 412) {
+                    promptUserWithModal(syncIssueTitle, noRemoteFileButtons, syncIssueFatalBody)
+                } else if (!res.ok && res.status >= 400) {
+                    throw Error(res.statusText)
+                }
+                return res.json()
+            })
+            .then(res => {
+                console.log('res in check meta after .json()', res)
+                handleCheckMetaResp(res) //sets meta state
+                toggleMetaFailureBanner(false) //sets banner for meta status
+                maintainLock(res) //gets lock 
+                return res
+            })
+            .catch(err => {
+                console.log('in checkMeta catch')
+                console.error(err)
+                removeElementById(modeBannerId)
+                toggleMetaFailureBanner(true)
+            })
+    }
+
+    //this function assumes any status not included in these lists represents an in sync notebook to defend against future fields being added being auto-categorized as failures
```

Comment:
It would be better to autocategorize future statuses as failures. LIVE should be the only good status for the foreseeable future. This is a good thing to call out though - luckily Welder and the UI extension are deployed in tandem so we should be pretty insulated against server/client skew in this case.